translation/templates.rst

```diff
@@ -0,0 +1,123 @@
+Using Translation in Templates
+==============================
+
+Twig Templates
+--------------
+
+.. _translation-tags:
+
+Using Twig Tags
+~~~~~~~~~~~~~~~
+
+Symfony provides specialized Twig tags (``trans`` and ``transchoice``) to
+help with message translation of *static blocks of text*:
+
+.. code-block:: twig
+
+    {% trans %}Hello %name%{% endtrans %}
+
+    {% transchoice count %}
+        {0} There are no apples|{1} There is one apple|]1,Inf[ There are %count% apples
+    {% endtranschoice %}
+
+The ``transchoice`` tag automatically gets the ``%count%`` variable from
+the current context and passes it to the translator. This mechanism only
+works when you use a placeholder following the ``%var%`` pattern.
+
+.. deprecated:: 4.2
+
+    The ``transchoice`` tag is deprecated since Symfony 4.2 and will be
+    removed in 5.0. Use the :doc:`ICU MessageFormat </translation/message_format>` with
+    the ``trans`` tag instead.
+
+.. caution::
+
+    The ``%var%`` notation of placeholders is required when translating in
+    Twig templates using the tag.
+
```

Comment:
Actually, that wasn't really required. You could also do `{var}` or something else (even just 'var' would work) :P

But it may be useful to add a warning about this not being compatible with the ICU version. That does not allow a variable named '%var%' and if you define `{var}` in a translation, it needs to be passed as `{'var': 'the content'}` rather than with percents.

I.e. if you want to solve the transchoice-deprecations, you must convert all translations to ICU-format.

This statement may also be necessary elsewhere or perhaps added to the default transchoice deprecation message.

Reply:
Yeah, the whole thing with translation tags/filters is a bit weird (the automatic var stuff). It's not really translation anymore, but just message formatting. I'm preparing a code PR to deprecate this magic and preparing a doc PR to remove these bad examples from the docs.